sdk/typescript/src/keypairs/ed25519/publickey.ts

```diff
@@ -73,4 +74,15 @@ export class Ed25519PublicKey extends PublicKey {
 	flag(): number {
 		return SIGNATURE_SCHEME_TO_FLAG['ED25519'];
 	}
+
+	/**
+	 * Verifies that the signature is valid for for the provided message
+	 */
+	async verify(message: Uint8Array, signature: Uint8Array): Promise<boolean> {
+		return nacl.sign.detached.verify(message, signature, this.toBytes());
+	}
+
+	static fromBytes(bytes: Uint8Array): Ed25519PublicKey {
+		return new Ed25519PublicKey(bytes);
```

Comment:
Hmm, I now realize that `from` still requires the constructor in this case... It feels less compelling to have `from` methods in these cases now